ITE/4/README.adoc

```diff
@@ -0,0 +1,283 @@
+must be updated.
+
+in-toto link metadata files currently look like this:
+
+```
+{
+    "_type": "link",
+    "_name": "<NAME>",
+    "command": "<COMMAND>",
+    "materials": {
+        "<PATH>": <HASH>,
+        "...": ...
+    },
+    "products": {
+        "<PATH>": <HASH>,
+        "...": ...
+    },
+    "byproducts": {
+        "stderr": "<STDERR>",
+        "stdout": "<STDOUT>",
+        "return-value": "<RETURN-VALUE>"
+    },
+    "environment": {
+        "variables": "<ENV>",
+        "filesystem": "<FS>",
+        "workdir": "<PWD>"
+    }
+}
+```
+
+The `materials` and `products` fields contain key-value pairs. The URI for each
+artifact is its path in the filesystem. With this ITE active, the artifact could
+instead have a generic URI which can be resolved to some external resource that
+is part of the supply chain. The URI MUST have a fixed structure and MUST
+contain all the information necessary for any verifier to obtain the resource it
+refers to. Therefore, any verifier MUST be able to take the URI in isolation and
+know how and where to find the artifact.
+
+This ITE proposes the following structure for the URI.
+
+`<scheme>:<hier-part>`
+
+This structure is derived from RFC 3986. The `scheme` defines the token or
+identifier for the type of resource the URI refers to . It can also contain
+other relevant information that indicates how the resource can be accessed. The
+`hier-part` identifies the path or location of the resource.
+
+A compliant in-toto system must be capable of generating link metadata with
+artifacts that have generic URI references. Therefore, this system must also be
+capable of resolving this URI and calculating cryptographic hashes of these
+artifacts. It's important to note that the contents of the resource are hashed,
+as opposed to the URI itself.
+
+To that end, it's important for a URI scheme to have a single mapping for the
+service it refers to. Consider an example of a GitHub specific resource - a pull
+request or an issue.
+
+`github+https://github.com/in-toto/in-toto/pull/250`
+
+This can also instead use GitHub's API specific URI to refer to the same data.
+
+`github+https://api.github.com/repos/in-toto/in-toto/pulls/250`
+
+The in-toto system generating link metadata for this artifact MUST be capable
+of recognizing that it possesses a generic URI, and the system MUST be able to
+resolve this resource and compute its cryptographic hash information. 
+
+An example of the product field for link metadata with a generic URI is as
+follows:
+
+```
+"products": {
+    "github+https://api.github.com/repos/in-toto/in-toto/pulls/250": {
+        "sha256": "7ee0617..."
+    },
+    "...": ...
+}
+```
+
+Similarly, an SPDX URI can be used to directly fetch the corresponding documents
+from a specified location. For in-toto metadata that exist within SPDX
+documents, the URI can also be used to refer to other entities elsewhere in the
+document.
```

Comment:
Is the reader of this ITE supposed to be familiar with SPDX and/or with in-toto/SPDX integration efforts? If not, some elaboration might be in order.

Why is an SPDX URI different form any other URI? And why is an SPDX URI that points to a resources in the same document different from any other SPDX URI?

Reply:
There's a bit about SPDX in motivation, and I'll expand that perhaps. As far as "different from any other URI", no, I don't see it being so. Let me work on the phrasing here.